### ISSUE TITLE
Error with socket.SO_REUSEPORT on WSL

### ISSUE THREAD

**Issue by Dakkaron** [eventlet/eventlet#418]
I found an error in eventlet running on WSL (Ubuntu on Windows):

```py
Traceback (most recent call last):
  File "ryu/bin/ryu-manager", line 21, in <module>
    main()
  File "/mnt/c/Users/dkryw/Documents/Arbeitsrepos/CubroControl/env/lib/python3.5/site-packages/ryu/cmd/manager.py", line 83, in main
    webapp = wsgi.start_service(app_mgr)
  File "/mnt/c/Users/dkryw/Documents/Arbeitsrepos/CubroControl/env/lib/python3.5/site-packages/ryu/app/wsgi.py", line 334, in start_service
    return WSGIServer(instance)
  File "/mnt/c/Users/dkryw/Documents/Arbeitsrepos/CubroControl/env/lib/python3.5/site-packages/ryu/app/wsgi.py", line 325, in __init__
    application, **config)
  File "/mnt/c/Users/dkryw/Documents/Arbeitsrepos/CubroControl/env/lib/python3.5/site-packages/ryu/lib/hub.py", line 123, in __init__
    self.server = eventlet.listen(listen_info)
  File "/mnt/c/Users/dkryw/Documents/Arbeitsrepos/CubroControl/env/lib/python3.5/site-packages/eventlet/convenience.py", line 58, in listen
    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
OSError: [Errno 92] Protocol not available
```

socket.SO_REUSEPORT is not supported on WSL and this is not likely to change (as seen here: https://github.com/Microsoft/BashOnWindows/issues/1419 and here: https://github.com/Microsoft/BashOnWindows/issues/1678 in the last post)

If I comment this line out, everything works fine. So I guess, the right solution would be to detect if Python is running on WSL and in that case skip that line.

Detecting WSL is quite easy, all you have to do is check if `"Microsoft"` is in `/prov/version`:

```
$ cat /proc/version
Linux version 4.4.0-43-Microsoft (Microsoft@Microsoft.com) (gcc version 5.4.0 (GCC) ) #1-Microsoft Wed Dec 31 14:42:53 PST 2014
```

**Comment by temoto** [comment]
Duplicate #380 
It's an upstream bug in CPython. If platform does not support REUSEPORT, then it should not be defined in `socket` module, that's how other socket options work.

**Comment by temoto** [comment]
Note for history: fix is merged in master 09e166bfbd6ba6f52477a7ae1c421dcabd1f004c
